agenta-backend/agenta_backend/services/evaluators_service.py

```diff
@@ -14,15 +13,79 @@
 
 from agenta_backend.services.security import sandbox
 from agenta_backend.models.shared_models import Error, Result
+from agenta_backend.models.api.evaluation_model import (
+    EvaluatorInputInterface,
+    EvaluatorOutputInterface,
+    EvaluatorMappingInputInterface,
+    EvaluatorMappingOutputInterface,
+)
 from agenta_backend.utils.traces import (
     process_distributed_trace_into_trace_tree,
     get_field_value_from_trace_tree,
 )
 
+
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
 
 
+async def map(
+    mapping_input: EvaluatorMappingInputInterface,
+) -> EvaluatorMappingOutputInterface:
+    """
+    Maps the evaluator inputs based on the provided mapping and data tree.
+
+    Returns:
+        EvaluatorMappingOutputInterface: A dictionary containing the mapped evaluator inputs.
+    """
+
+    async def get_nested_value(data: Dict[str, Any], key: str) -> Any:
```

Comment:
Why do we still have this instead of using `utils.traces` ?

Reply:
I've replaces get_nested_value with it, thank you.